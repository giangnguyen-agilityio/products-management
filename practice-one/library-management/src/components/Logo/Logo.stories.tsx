import { type StoryFn, type Meta } from "@storybook/react";
import Logo from "./index";

export default {
  title: "Example/Logo",
  component: Logo,
} as Meta<typeof Logo>;

const Template: StoryFn<typeof Logo> = (args) => <Logo {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  imageSrc: "src/assets/images/logo-website.png",
  altText: "This is the logo website",
  widthSize: 50,
  heightSize: 50,
};

export const Secondary = Template.bind({});
Secondary.args = {
  text: "Logo Website",
};
