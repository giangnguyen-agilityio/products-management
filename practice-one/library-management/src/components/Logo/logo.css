.logo {
  font-size: 26px;
  font-family: var(--primary-font);
  color: var(--primary-color);
}
