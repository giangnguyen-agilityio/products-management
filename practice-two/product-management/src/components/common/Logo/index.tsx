import React from 'react'
import { Flex, Heading, Image } from '@chakra-ui/react'

interface LogoProps {
  text?: string
  textColor?: string
  textSize?: string
  imageSrc?: string
  widthSize?: string | number
  heightSize?: string | number
}

const Logo: React.FC<LogoProps> = ({
  text,
  textColor = 'black',
  textSize = 'md',
  imageSrc,
  widthSize = '120px',
  heightSize = '40px',
}) => {
  return (
    <Flex align="center" className="logo" cursor="pointer">
      {/* Render the text if provided */}
      {text && (
        <Heading as="h1" size={textSize} color={textColor}>
          {text}
        </Heading>
      )}
      {/* Render the image if provided */}
      {imageSrc && (
        <Image
          className="logo-image"
          loading="eager"
          srcSet={imageSrc}
          src={imageSrc}
          alt="The logo website"
          width={widthSize}
          height={heightSize}
        />
      )}
    </Flex>
  )
}

export default Logo
