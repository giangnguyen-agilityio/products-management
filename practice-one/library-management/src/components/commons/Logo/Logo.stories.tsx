import {type StoryFn, type Meta} from '@storybook/react'
import {BrowserRouter} from 'react-router-dom'
import Logo from './index'
import React from 'react'

export default {
  title: 'Example/Logo',
  component: Logo,
} as Meta<typeof Logo>

const Template: StoryFn<typeof Logo> = args => (
  <BrowserRouter>
    <Logo {...args} />
  </BrowserRouter>
)

export const Primary = Template.bind({})
Primary.args = {
  imageSrc: 'src/assets/images/logo-website.png',
  altText: 'This is the logo website',
  widthSize: 50,
  heightSize: 50,
}

export const Secondary = Template.bind({})
Secondary.args = {
  text: 'Logo Website',
}
