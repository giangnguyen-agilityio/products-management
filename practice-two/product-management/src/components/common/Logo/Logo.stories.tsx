import type { Meta, StoryObj } from '@storybook/react'
import Logo from './index'

const meta = {
  title: 'Practice Two/Logo Component',
  component: Logo,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    textColor: { control: 'color' },
  },
} satisfies Meta<typeof Logo>

export default meta
type Story = StoryObj<typeof meta>

export const LogoWithText: Story = {
  args: {
    text: 'Agility',
    textSize: 'md',
  },
}

export const LogoWithImage: Story = {
  args: {
    imageSrc: 'src/assets/images/logo_website.webp',
    widthSize: '100px',
    heightSize: '40px',
  },
}
