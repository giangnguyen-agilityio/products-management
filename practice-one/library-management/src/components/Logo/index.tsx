import './logo.css'

interface LogoProps {
  text?: string
  imageSrc?: string
  altText?: string
  size?: number
}

const Logo = (props: LogoProps) => {
  const { text, imageSrc, altText, size } = props
  return (
    <div className="logo">
      {(text != null) && <h1>{text}</h1>}
      {(imageSrc != null) && (
        <img
          src={imageSrc}
          alt={altText}
          style={{ width: `${size ?? 50}px`, height: `${size ?? 50}px` }}
        />
      )}
    </div>
  )
}

export default Logo
