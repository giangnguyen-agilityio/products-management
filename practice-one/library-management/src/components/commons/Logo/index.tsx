import React from 'react'
import {Link} from 'react-router-dom'
import {getItemInLocalStorage} from '@helpers'
import './logo.css'

interface LogoProps {
  text?: string
  imageSrc?: string
  altText?: string
  widthSize?: number
  heightSize?: number
}

const Logo: React.FC<LogoProps> = props => {
  const isMember: string = getItemInLocalStorage('memberId')

  const {text, imageSrc, altText, widthSize, heightSize} = props
  return isMember ? (
    <Link to="/">
      <div className="logo">
        {/* Render the text if provided */}
        {text && <h1>{text}</h1>}
        {/* Render the image if provided */}
        {imageSrc && (
          <img
            src={imageSrc}
            alt={altText}
            style={{
              width: `${widthSize ?? 50}px`,
              height: `${heightSize ?? 50}px`,
            }}
          />
        )}
      </div>
    </Link>
  ) : (
    <div className="logo">
      {/* Render the text if provided */}
      {text && <h1>{text}</h1>}
      {/* Render the image if provided */}
      {imageSrc && (
        <img
          src={imageSrc}
          alt={altText}
          style={{
            width: `${widthSize ?? 50}px`,
            height: `${heightSize ?? 50}px`,
          }}
        />
      )}
    </div>
  )
}

export default Logo
