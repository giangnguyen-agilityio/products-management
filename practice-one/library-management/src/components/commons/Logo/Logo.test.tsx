import React from 'react'
import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import * as helpers from '@helpers'
import Logo from '.'

// Mocking the module containing getItemInLocalStorage
jest.mock('@helpers', () => ({
  getItemInLocalStorage: jest.fn(),
}))

describe('Logo component', () => {
  const logoProps = {
    text: 'Test Logo',
    imageSrc: 'test-image.png',
    altText: 'Test Alt Text',
    widthSize: 100,
    heightSize: 100,
  }

  it('should renders with Link when isMember is truthy', () => {
    // Mocking getItemInLocalStorage to return a truthy value
    jest.spyOn(helpers, 'getItemInLocalStorage').mockReturnValue('M01')

    const {container, getByText} = render(
      <MemoryRouter>
        <Logo {...logoProps} />
      </MemoryRouter>
    )

    // The component should render with <Link>
    expect(container.querySelector('a')).toBeInTheDocument()
    expect(container.querySelector('.logo')).toBeInTheDocument()
    expect(getByText('Test Logo')).toBeInTheDocument()
    expect(container.querySelector('img')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should renders without Link when isMember is falsy', () => {
    // Mocking getItemInLocalStorage to return a falsy value
    jest.spyOn(helpers, 'getItemInLocalStorage').mockReturnValue(null)

    const {container} = render(
      <MemoryRouter>
        <Logo {...logoProps} />
      </MemoryRouter>
    )

    // The component should not render with <Link>
    expect(container.querySelector('a')).not.toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should render the image with the default width and height when not provided', () => {
    const {container} = render(
      <MemoryRouter>
        <Logo
          text="Test Logo"
          imageSrc="test-image.png"
          altText="Test Alt Text"
        />
      </MemoryRouter>
    )

    // The component should render with default width and height of 50px
    const logoImage = container.querySelector('img')
    expect(container).toMatchSnapshot()
    expect(logoImage?.style.width).toBe('50px')
    expect(logoImage?.style.height).toBe('50px')
  })
})
