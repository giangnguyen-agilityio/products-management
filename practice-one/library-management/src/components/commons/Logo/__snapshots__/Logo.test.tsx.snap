// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component should render the image with the default width and height when not provided 1`] = `
<div>
  <div
    class="logo"
  >
    <h1>
      Test Logo
    </h1>
    <img
      alt="Test Alt Text"
      src="test-image.png"
      style="width: 50px; height: 50px;"
    />
  </div>
</div>
`;

exports[`Logo component should renders with Link when isMember is truthy 1`] = `
<div>
  <a
    href="/"
  >
    <div
      class="logo"
    >
      <h1>
        Test Logo
      </h1>
      <img
        alt="Test Alt Text"
        src="test-image.png"
        style="width: 100px; height: 100px;"
      />
    </div>
  </a>
</div>
`;

exports[`Logo component should renders without Link when isMember is falsy 1`] = `
<div>
  <div
    class="logo"
  >
    <h1>
      Test Logo
    </h1>
    <img
      alt="Test Alt Text"
      src="test-image.png"
      style="width: 100px; height: 100px;"
    />
  </div>
</div>
`;
